prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -179,6 +192,121 @@ def filter_by_connected_components(lines, substations):
     return remaining_lines, remaining_substations
 
 
+def augment_line_voltages(
+    lines, substations, volt_class_defaults=None, state_threshold_100_161=0.95
+):
+    """Fill in voltages for lines with missing voltages, using a series of heuristics.
+    The ``lines`` dataframe will be modified in-place.
+
+    :param pandas.DataFrame lines: data frame of lines.
+    :param pandas.DataFrame substations: data frame of substations.
+    :param dict/pandas.Series volt_class_defaults: mapping of volt classes to default
+        replacement voltage values. If None, internally-defined defaults will be used.
+    :param int/float state_threshold_100_161: fraction of lines in the 100-161 kV range
+        which must be the same voltage for the most common voltage to be applied to
+        missing voltage lines in this state.
+    """
+
+    # Interpret input parameters
+    if volt_class_defaults is None:
+        volt_class_defaults = {
+            "UNDER 100": 69,
+            "220-287": 230,
+            "345": 345,
+            "500": 500,
+        }
+
+    # Replace dummy data with explicit 'missing'
+    lines.loc[lines.VOLTAGE == -999999, "VOLTAGE"] = pd.NA
```

Comment:
should we move this in the `load` module?

Reply:
I think that would be fine.

Reply:
Done.